pkg/controller/cache/managed.go

```diff
@@ -100,19 +102,19 @@ func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (resource.
 	if c.newClientFn != nil {
 		newClientFn = c.newClientFn
 	}
```

Comment:
Nitpick - it would be nice to set `c.newClientFn` inside `SetupWithManager` as we did with the GCP updates.